### PR TITLE
feat(detection): use numerical stable focal loss and EMA normalizer in RetinaNet

### PR DESCRIPTION
**Pull request by wjfwzzc**

### CODE REVIEW

File: official/vision/detection/models/retinanet.py [hunk 1]
```diff
@@ -104,7 +106,7 @@ def forward(self, inputs):
             for i in range(len(features))
         ]
 
-        all_level_box_cls = F.sigmoid(F.concat(box_cls_list, axis=1))
+        all_level_box_cls = F.concat(box_cls_list, axis=1)
```
Comment (FateScript) [inline official/vision/detection/models/retinanet.py:109]: 这个地方要不叫all_level_box_logits?

File: official/vision/detection/tools/test.py [hunk 2]
```diff
@@ -55,9 +55,16 @@ def main():
...
     sys.path.insert(0, os.path.dirname(args.file))
     current_network = importlib.import_module(os.path.basename(args.file).split(".")[0])
+    cfg = current_network.Cfg()
 
-    if args.end_epoch == -1:
-        args.end_epoch = args.start_epoch
+    if args.weight_file:
+        args.start_epoch = args.end_epoch = -1
+    else:
+        if args.start_epoch == -1:
```
Comment (FateScript) [inline official/vision/detection/tools/test.py:63]: 按说这个负索引的逻辑应该不只是-1能work吧，直接cfg.max_epoch + start_epoch不行？

File: official/vision/detection/models/retinanet.py [hunk 3]
```diff
@@ -114,22 +116,35 @@ def forward(self, inputs):
...
                 )
                 * self.cfg.reg_loss_weight
             )
 
+            if norm_type == "none":
+                F.add_update(
+                    self.loss_normalizer,
+                    (box_gt_cls > 0).sum(),
+                    alpha=self.cfg.loss_normalizer_momentum,
+                    beta=1 - self.cfg.loss_normalizer_momentum,
```
Comment (FateScript) [inline official/vision/detection/models/retinanet.py:143]: 这个减号前后应该不需要空格吧

File: official/vision/detection/tools/test.py (hunk 2 above)
Comment (wjfwzzc) [reply official/vision/detection/tools/test.py:63]: 这个负数只是起到一个默认参数的作用，我们不支持负索引。

File: official/vision/detection/models/retinanet.py (hunk 3 above)
Comment (wjfwzzc) [reply official/vision/detection/models/retinanet.py:143]: 这个和之前那个除号是同一个问题呀

File: official/vision/detection/models/retinanet.py (hunk 1 above)
Comment (wjfwzzc) [reply official/vision/detection/models/retinanet.py:109]: 这个我想过，主要是faster r-cnn也叫cls。过了softmax之后叫score

Comment (wjfwzzc) [reply official/vision/detection/models/retinanet.py]: 大面积修了一把。
